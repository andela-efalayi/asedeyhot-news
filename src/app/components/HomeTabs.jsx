/**
 * @function HomeTabs
 * @return {object} react-componet
 */

import React from 'react';
import { Tabs, Tab } from 'material-ui/Tabs';
import FavoriteBorder from 'material-ui/svg-icons/action/favorite-border';
import Public from 'material-ui/svg-icons/social/public';

const changeTab = (tab) => {
  window.location = tab.props['data-route'];
};

const HomeTabs = () => {
  return (
    <Tabs className="home-tabs">
      <Tab icon={<Public />} data-route="#/sources"
      label="sources" onActive={changeTab}/>
      <Tab icon={<FavoriteBorder />} data-route="#/favourites"
      label="Favorites" onActive={changeTab}/>
    </Tabs>
  );
};

export default HomeTabs;
