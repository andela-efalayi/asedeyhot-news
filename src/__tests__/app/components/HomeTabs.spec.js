import React from 'react';
import { shallow } from 'enzyme';
import HomeTabs from '../../../app/components/HomeTabs.jsx';

describe('HomeTabs Component', () => {
  it('should render hometabs', () => {
    const wrapper = shallow(
        <HomeTabs />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
