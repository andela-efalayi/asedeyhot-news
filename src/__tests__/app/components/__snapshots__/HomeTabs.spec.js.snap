// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTabs Component should render hometabs 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Tabs
    initialSelectedIndex={0}
    onChange={[Function]}
>
    <Tab
        data-route="#/sources"
        icon={<SocialPublic />}
        label="sources"
        onActive={[Function]}
    />
    <Tab
        data-route="#/favourites"
        icon={<ActionFavoriteBorder />}
        label="Favorites"
        onActive={[Function]}
    />
</Tabs>,
  "nodes": Array [
    <Tabs
      initialSelectedIndex={0}
      onChange={[Function]}
>
      <Tab
            data-route="#/sources"
            icon={<SocialPublic />}
            label="sources"
            onActive={[Function]}
      />
      <Tab
            data-route="#/favourites"
            icon={<ActionFavoriteBorder />}
            label="Favorites"
            onActive={[Function]}
      />
</Tabs>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <HomeTabs />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Tabs
          initialSelectedIndex={0}
          onChange={[Function]}
>
          <Tab
                    data-route="#/sources"
                    icon={<SocialPublic />}
                    label="sources"
                    onActive={[Function]}
          />
          <Tab
                    data-route="#/favourites"
                    icon={<ActionFavoriteBorder />}
                    label="Favorites"
                    onActive={[Function]}
          />
</Tabs>,
        "_debugID": 2,
        "_renderedOutput": <Tabs
          initialSelectedIndex={0}
          onChange={[Function]}
>
          <Tab
                    data-route="#/sources"
                    icon={<SocialPublic />}
                    label="sources"
                    onActive={[Function]}
          />
          <Tab
                    data-route="#/favourites"
                    icon={<ActionFavoriteBorder />}
                    label="Favorites"
                    onActive={[Function]}
          />
</Tabs>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HomeTabs />,
}
`;
